src/shogun/machine/gp/SingleFITCLaplacianBase.cpp

```diff
@@ -0,0 +1,409 @@
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * The views and conclusions contained in the software and documentation are those
+ * of the authors and should not be interpreted as representing official policies,
+ * either expressed or implied, of the Shogun Development Team.
+ *
+ */
+
+#include <shogun/machine/gp/SingleFITCLaplacianBase.h>
+
+#ifdef HAVE_EIGEN3
+
+#include <shogun/mathematics/Math.h>
+#include <shogun/mathematics/eigen3.h>
+#include <shogun/features/DotFeatures.h>
+
+using namespace shogun;
+using namespace Eigen;
+
+CSingleFITCLaplacianBase::CSingleFITCLaplacianBase() : CFITCInferenceBase()
+{
+	init();
+}
+
+CSingleFITCLaplacianBase::CSingleFITCLaplacianBase(CKernel* kern, CFeatures* feat,
+		CMeanFunction* m, CLabels* lab, CLikelihoodModel* mod, CFeatures* lat)
+		: CFITCInferenceBase(kern, feat, m, lab, mod, lat)
+{
+	init();
+	check_fully_FITC();
+}
+
+void CSingleFITCLaplacianBase::init()
+{
+	m_fully_FITC=false;
+
+	SG_ADD(&m_al, "al", "alpha", MS_NOT_AVAILABLE);
+	SG_ADD(&m_t, "t", "noise", MS_NOT_AVAILABLE);
+	SG_ADD(&m_B, "B", "B", MS_NOT_AVAILABLE);
+	SG_ADD(&m_w, "w", "B*al", MS_NOT_AVAILABLE);
+	SG_ADD(&m_Rvdd, "Rvdd", "Rvdd", MS_NOT_AVAILABLE);
+	SG_ADD(&m_V, "V", "V", MS_NOT_AVAILABLE);
+	SG_ADD(&m_fully_FITC, "fully_FITC", "whether the kernel support fitc inference", MS_NOT_AVAILABLE);
+}
+
+void CSingleFITCLaplacianBase::set_kernel(CKernel* kern)
+{
+	CInferenceMethod::set_kernel(kern);
+	check_fully_FITC();
+}
+
+CSingleFITCLaplacianBase::~CSingleFITCLaplacianBase()
+{
+}
+
+void CSingleFITCLaplacianBase::check_fully_FITC()
+{
+	REQUIRE(m_kernel, "Kernel must be set first\n")
+	if (strstr(m_kernel->get_name(), "FITCKernel")!=NULL)
+	{
```

Comment:
currently we check the name of the provided kernel to determine whether the kernel can compute the derivatives wrt inducing_features.
The reason is the base class `CKernel` does not have the `is_support_FITC()`.
The `wrapped kernel` approach is hard to implement since the static template methods in `CKernel`.